src/components/Navigation.tsx

```diff
@@ -20,11 +24,30 @@ const Navigation: FC = () => {
   const [projectName, setProjectName] = useState(
     project && !isLoading ? project.name : "default",
   );
+  const { hasCustomVolumeIso } = useSupportedFeatures();
+  const enablePermissions = enablePermissionsFeature();
+  const [scroll, setScroll] = useState(false);
+  const [openNavMenus, setOpenNavMenus] = useState<Set<AccordionNavMenus>>(
+    new Set(),
+  );
 
   useEffect(() => {
     project && project.name !== projectName && setProjectName(project.name);
   }, [project?.name]);
 
+  useEffect(() => {
+    if (!menuCollapsed) {
+      adjustNavigationScrollForOverflow();
+    } else {
+      setScroll(false);
+      setOpenNavMenus(new Set());
```

Comment:
It think, we should avoid this in case the state is already an empty set.

We should also avoid the setScroll calls above and below, if there is no change. I assume all of them would otherwise cause another render cycle due to changed state (that didn't really need to be triggered).

Reply:
This should do the trick

 ```
 useEffect(() => {
    if (!menuCollapsed) {
      adjustNavigationScrollForOverflow();
      return;
    }

    if (scroll) {
      setScroll(false);
    }

    if (openNavMenus.size) {
      setOpenNavMenus(new Set());
    }
  }, [menuCollapsed, scroll, openNavMenus]);
```